src/components/UserProfileSettings.tsx

```diff
@@ -0,0 +1,133 @@
+import { Button } from '@/components/ui/button'
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from '@/components/ui/tooltip'
+import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar'
+import { PaywallDialog } from '@/components/PaywallDialog'
+import { GoogleIcon } from '@/components/icons/GoogleIcon'
+import { LogOut, KeyRound } from 'lucide-react'
+import { format } from 'date-fns'
+import { User } from '@supabase/supabase-js'
+import { useLicenseStore } from '@/stores/licenseStore'
+import supabase from '@/lib/integrations/supabase'
+import { useNavigate } from 'react-router-dom'
+import { info as logInfo, warn as logWarn, error as logError } from '@tauri-apps/plugin-log'
+
+const DEVICE_ID_KEY = 'ebb_device_id'
+
+interface UserProfileSettingsProps {
+  user: User | null
+}
+
+export function UserProfileSettings({ user }: UserProfileSettingsProps) {
+  const navigate = useNavigate()
+  const license = useLicenseStore((state) => state.license)
+  const isLoadingLicense = useLicenseStore((state) => state.isLoading)
+
+  const hasProLicense = !isLoadingLicense && license && (license.status === 'active' || license.status === 'trialing')
```

Comment:
ditto from before about being able to put this in the store,

Also, I believe you can reference the store like this and destructure the elements out of it like so: 

```
const { isLoadingLicense, hasProLicense } = useLicenseStore();
```